### ISSUE TITLE
Inputs to memristive layers after convolutional ones are `nan`

### ISSUE THREAD

**Issue by joksas** [joksas/nonideality-aware-mnn-training#3]
If in
https://github.com/awai54st/Non-Idealility-Aware-MBNN-Training/blob/f0c4036bab00efa3899deab87f8d87b78073bbc9/training/architecture.py#L24
we change `activation` from `sigmoid` to `relu`, the inputs to the first memristive layer after it become `nan`, but only in the second batch of the first training epoch -- the first batch is always fine no matter what the batch size is! It's as if the weights are changed so much in the first batch that the second batch produces infinities.

For example, if we uncomment
https://github.com/awai54st/Non-Idealility-Aware-MBNN-Training/blob/f0c4036bab00efa3899deab87f8d87b78073bbc9/training/architecture.py#L112-L113
the output is something like
```text
x shape: TensorShape([100, 1024])
x: [[0.0745160282 0.129433334 0.0827285051 ... 0 0.0890432745 0.177923]
 [0.110229135 0.139877185 0 ... 0 0.0779622272 0.122682653]
 [0.13858591 0.0414419919 0 ... 0 0.138552755 0.211386681]
 ...
 [0.0930541307 0.0710040182 0 ... 0 0.2002794 0.229018793]
 [0.102769464 0.0778024197 0 ... 0 0.153720185 0.244958788]
 [0.0918904543 0.0298978798 0 ... 0.00886781886 0.132527441 0.192799985]]
```
in the first batch (in this case, 100 examples), and
```text
x shape: TensorShape([100, 1024])
x: [[-nan -nan -nan ... -nan 0 0]
 [-nan -nan -nan ... -nan 0 0]
 [-nan -nan -nan ... -nan 0 0]
 ...
 [-nan -nan -nan ... -nan 0 0]
 [-nan -nan -nan ... -nan 0 0]
 [-nan -nan -nan ... -nan 0 0]]
```
in the second batch. Sometimes there are `inf`s (or something like `2e+38`) instead of `nan`s.

This is really weird. And changing `relu` to `sigmoid` isn't a great solution -- there might be some bug that I'm overlooking.

I was thinking that the convolutional layers were somehow producing infinities on their own, but simply changing first memristive layer
https://github.com/awai54st/Non-Idealility-Aware-MBNN-Training/blob/f0c4036bab00efa3899deab87f8d87b78073bbc9/training/architecture.py#L28
to keras's traditional dense layer
```python
model.add(layers.Dense(num_hidden_neurons))
```
eliminates the problem. So it seems that the issue lies not in the preceding layers, but in `MemristorDense`.
